dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/v34/V2_34_22__Add_teav_btree_index.java

```diff
@@ -0,0 +1,46 @@
+package org.hisp.dhis.db.migration.v34;
+
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import static org.hisp.dhis.trackedentity.TrackedEntityAttributeService.TEA_VALUE_MAX_LENGTH;
+
+/**
+ * This Java migration changes the data type of the 'value' column in trackedentityattributevalue table to varchar(1200).
+ * Additionally we put a btree index on the same table, to improve performance of lookup.
+ * This upograde will fail is any value in the table exceeds 1200 characters. Due to this, we catch any SQLException and
+ * write a custom message, including a link to a community.dhis2.org post explaining this upgrade, and how to deal with any
+ * upgrade failures.
+ *
+ * @author Stian
+ */
+public class V2_34_22__Add_teav_btree_index
+    extends BaseJavaMigration
+{
+    private final static String COP_POST_URL = "https://community.dhis2.org/t/draft-important-database-upgrade-for-tracker-performance/38766";
```

Comment:
Is this URL available right now? Or is it private? When I tried it said "Page doesnt exist, or is private"

Reply:
That URL points to a draft that is locked in COP. I will update the URL when the post have been published and gotten a final URL :)